integration/src/test/resources/org/mule/test/components/redelivery-policy-config.xml

```diff
@@ -95,4 +114,35 @@
             </on-error-propagate>
         </error-handler>
     </flow>
+
+    <flow name="redeliveryInvalidMessageIdWithTransactionalSourceAndCustomErrorHandlerDispatch">
+        <vm:publish-consume queueName="Q5" config-ref="VM_Config"/>
+    </flow>
+
+    <flow name="redeliveryInvalidMessageIdWithTransactionalSourceAndCustomErrorHandlerProcess">
+        <vm:listener queueName="Q5" config-ref="VM_Config" transactionalAction="ALWAYS_BEGIN">
+            <redelivery-policy idExpression="payload.id ++ '-' ++ payload.id"/>
+        </vm:listener>
+
+        <logger/>
+
+        <error-handler>
+            <on-error-propagate type="EXPRESSION">
+                <set-payload value="#[error]"/>
```

Comment:
use `#[!null.a]` expression to force an expression error

Reply:
I didn't intend to force an expression error there, just get the error itself for the log. However I did change it in the two places where I needed to force an expression error.